content/tutorials/react-admin-panel-with-ably-and-refine.textile

```diff
@@ -0,0 +1,289 @@
+
+```[sh]
+npm install @refinedev/ably
+```
+
+h3. Step 3 — Integrate Ably with refine
+
+After successfully installing the __@refinedev/ably package__ in the previous step, integrate Ably with refine. Start by creating a @.env@ file at the root of your refine project directory for keeping your Ably API key.
+
+Add the following environment variable to the @.env@ file and set its value to your Ably API key. Replace @1234@ with your Ably API key. Be sure to restart the development server after creating an environment variable.
+
+```[sh]
+REACT_APP_ABLY_API_KEY=1234
+```
+
+Add the @.env@ file you have created to your @.gitignore@ file so you don’t expose it in a version control system like Git.
+
+Create the @src/utility/ablyClient.ts@ and @src/utility/index.ts@ files. Copy and paste the code below into the @src/utility/ablyClient.ts@ you have created.
+
+The code below uses your Ably API key to create an Ably client and exports it to use anywhere in your refine application.
+
+```[nodejs]
+import { Ably } from "@refinedev/ably";
+
+export const ablyClient = new Ably.Realtime(process.env.REACT_APP_ABLY_API_KEY!);
+```
+
+Similarly, copy and paste the code below into the @src/utility/index.ts@ file you created. It will enable you to import exports from the @src/utility@ directory.
+
+```[nodejs]
+export * from "./ablyClient";
+```
+
+The @<Refine />@ component is the entry point for any refine application. The App component renders the @<Refine />@ component in the @App.tsx@ file. The @liveProvider@ prop of the @<Refine />@ component is for integrating refine with any realtime provider.
+
+The @liveProvider@ prop is agnostic of any realtime solution. You can use it with any realtime provider. In this article, you will use it to integrate Ably with your refine application. Apply the changes below to your @App.tsx@ file.
+
+```[nodejs]
+...
+//highlight-start
+import { ablyClient } from "utility";
+import { liveProvider } from "@refinedev/ably";
+//highlight-end
+
+function App() {
+  return (
+          ...
+            <Refine
+              ...
+              //highlight-next-line
+              liveProvider={liveProvider(ablyClient)}
+              options={{
+              ...
+                //highlight-next-line
+                liveMode: "auto",
+              }}
+            >
+            ...
+  );
+}
+
+export default App;
+```
+
+h3. liveMode
+
+Quick info for live mode options.
+
+@auto@: Queries of related resource are invalidated in Realtime as new events from subscription arrive.
+
+@manual@: Queries of related resource are not invalidated in Realtime, instead "onLiveEvent":https://refine.dev/docs/api-reference/core/providers/live-provider/#onliveevent is run with the event as new events from subscription arrive.
+
+@off@: Disables live mode.
+
+"Refer to live provider docs for more information":https://refine.dev/docs/api-reference/core/providers/live-provider/
+
+h3. We are ready for testion realtime feature on the app
```

Comment:
>testion ?

I'm not sure what this heading is trying to say given the content below it.

Reply:
Hmmm me neither, I'll update it to something like 'Testing out the app', as it seems to be a section on just seeing it in action.